### PR TITLE
user

### PR DESCRIPTION
**Pull request by YaTut1901**

### CODE REVIEW

File: src/main/java/main/movie/dao/impl/UserDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    public Optional<User> getByEmail(String email) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+            CriteriaQuery<User> query = criteriaBuilder.createQuery(User.class);
+            Root<User> root = query.from(User.class);
+            Predicate emailPredicate = criteriaBuilder.equal(root.get("email"), email);
+            return session.createQuery(query.where(emailPredicate)).uniqueResultOptional();
+        }
+    }
+}
```
Comment (tarkaiev) [inline src/main/java/main/movie/dao/impl/UserDaoImpl.java:51]: we didn't have to implement getall and get methods here
just findByEmail (wrong naming) and add (not create)

File: src/main/java/main/movie/security/impl/AuthenticationServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        }
+        throw new AuthenticationException("Wrong login for that user!");
+    }
+
+    @Override
+    public User register(String email, String password) {
+        User user = new User();
+        user.setEmail(email);
+        user.setSalt(HashUtil.generateSalt());
+        user.setPassword(HashUtil.hashPassword(password, user.getSalt()));
```
Comment (tarkaiev) [inline src/main/java/main/movie/security/impl/AuthenticationServiceImpl.java]: we have to do hashing and salting password in userservice

Comment (RomanDubovskyi) [reply src/main/java/main/movie/security/impl/AuthenticationServiceImpl.java]: > we have to do hashing and salting password in userservice

\+

File: src/main/java/main/movie/security/impl/AuthenticationServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> optionalUser = userService.findByEmail(email);
+        if (optionalUser.isEmpty()) {
```
Comment (RomanDubovskyi) [inline src/main/java/main/movie/security/impl/AuthenticationServiceImpl.java]: let's collapse to if clauses into one

File: src/main/java/main/movie/dao/impl/UserDaoImpl.java (hunk 1 above)
Comment (RomanDubovskyi) [reply src/main/java/main/movie/dao/impl/UserDaoImpl.java:51]: > we didn't have to implement getall and get methods here
> just findByEmail (wrong naming) and add (not create)

yes, let's go step by step

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**